core/src/main/scala/cats/AutoVariance.scala

```diff
@@ -0,0 +1,13 @@
+package cats
+
+trait AutoVariance extends AutoVarianceLowPriority {
+  implicit def autoWidenFunctor[F[_]: Functor, A, B >: A](fa: F[A]): F[B] = Functor[F].widen(fa)
+
+  implicit def autoNarrowContravariant[F[_]: Contravariant, A, B <: A](fa: F[A]): F[B] = Contravariant[F].narrow(fa)
+
+}
+
+trait AutoVarianceLowPriority {
+  implicit def autoWidenBifunctor[F[_, _]: Bifunctor, A, B >: A, C, D >: C](fac: F[A, C]): F[B, D] =
+    Bifunctor[F].leftWiden(Bifunctor[F].rightFunctor.widen(fac))
```

Comment:
Actually, random thought: should *this* be the high priority one and the others the low priority? Since this is the more specific coercion? I feel like I'm tying myself in mental knots here, but your test seems to indicate that it works fine as-written.

Reply:
Is there even a more specific coercion between these two? I feel like that would apply if `Functor` and `Bifunctor` had some kind of subtype relationship, but since they don't, I'm not sure

Reply:
Well, I'm just thinking about the situation with `Either` or `Tuple2`. We want `(String, String)` to coerce up to `(AnyRef, AnyRef)` if necessary, rather than preferring to coerce to `(String, AnyRef)`.

Reply:
I just checked and it seems like both version work, I'll change it to make the bifunctor one more specific though.

Reply:
Just when you think you understand scalac… I would have thought that the compiler would prefer the simpler coercion to the expense of the compound one.

Reply:
There are two things that matter - subtyping relationship of the return type and subtyping relationship of the defining class. Here we have only the latter I guess.